src/Tests/Core.Tests/CommandTests.cs

```diff
@@ -520,4 +521,40 @@ private async Task<IWorld> PrepareWorldAsync()
         await world.Load();
         return world;
     }
+    
+    [Fact]
+    public async Task ReloadPermissionsCommand_WithoutTarget()
+    {
+        // Prepare
+        var updatedGroup = Guid.NewGuid();
+        var groupName = "test";
+        
+        var newPermissions = new[] { "reload_perms", "goto" };
+        _services.GetRequiredService<ICommandPermissionRepository>().GetPermissionsForGroupAsync(Arg.Any<Guid>()).Returns(newPermissions);
+        _services.GetRequiredService<ICommandPermissionRepository>().GetGroupsAsync().Returns(new[] { (updatedGroup, groupName) });
+        
+        _services.GetRequiredService<ICacheManager>().CreateList<Guid>(Arg.Any<string>())
+            .Range(0, 0).Returns(new[] { updatedGroup });
+        
+        // Act
+        await _commandManager.Handle(_connection, "/reload_perms");
+
+        // Assert
+        _commandManager.Groups.Keys.Should().Contain(updatedGroup);
+        _commandManager.Groups.Values.Should().ContainEquivalentOf(new PermissionGroup
+        {
+            Id = updatedGroup,
+            Name = groupName,
+            Permissions = newPermissions
+        });
+        
+        _player.ReloadPermissions().Should().Be(Task.CompletedTask);
```

Comment:
This may fail if the task is not completed fast enough?
Also the exception may be hidden because this just checks if the result is completed. It maybe smarter to just `await` the call and let the test crash if anything happens.

Reply:
Yep ended up doing that, but I'm going to still add at least another case with players in the world so that we can actually assert then final groups of said players.